MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,9 +1,159 @@
+    private var currentQuestionIndex = 0
+    private var correctAnswers: Int = 0
+    private let questionsAmount: Int = 10
+    private var questionFactory: QuestionFactoryProtocol?
+    private var currentQuestion: QuizQuestion?
+    private var delegate: AlertPresenterDelegate?
+    private var documentsURL = Bundle.main.url(forResource: "top250MoviesIMDB", withExtension: "json")
+    private var statisticService: StatisticService?
+  
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        questionFactory = QuestionFactory(delegate: self)
+        questionFactory?.requestNextQuestion()
+        delegate = AlertPresenter(delegate: self)
+        statisticService = StatisticServiceImplementation()
+        guard let documentsURL = documentsURL else {return}
+        let jsonString = try! String(contentsOf: documentsURL)
+        let data = jsonString.data(using: .utf8)!
+      
+        do {
+            let movie = try JSONDecoder().decode(Top.self, from: data)
+            print(movie)
+        } catch {
+            print("Failed to parse: \(error.localizedDescription)")
+        }
+    }
+    
+    // MARK: - QuestionFactoryDelegate
+    func didRecieveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {
+            return
+        }
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async {[weak self] in
+            self?.show(quiz: viewModel)
+        }
+    }
+    
+    // MARK: - AlertPresenterDelegate
+    func updateGameCounter() {
+        //Обнуляем счетчик вопросов
+        self.currentQuestionIndex = 0
+        // Обнуляем счетчик правельных ответов
+        self.correctAnswers = 0
+        self.counterLabel.text = "\(self.correctAnswers)/10"
+        // заново показываем первый вопрос
+        self.questionFactory?.requestNextQuestion()
+    
+    }
+
+    func show(quiz result: QuizResultsViewModel) {
+        guard let statisticService = statisticService else {return}
+        self.statisticService?.gamesCount += 1
+        self.statisticService?.store(correct: correctAnswers, total: 10)
+
+        let alertModel = AlertModel(title: result.title,
+                                    message: """
+                                    Ваш результат: \(correctAnswers)/10\n
+                                    Количество сыгранных квизов: \(statisticService.gamesCount)\n
+                                    Рекорд: \(statisticService.bestGame.correct)/10 (\(statisticService.bestGame.date.dateTimeString))\n
+                                    Средняя точнось: \(String(format: "%.2f", statisticService.totalAccuracy))%"
+
+                                    
+                                    """,
+                                    buttonText: result.buttonText)
+        guard let delegate = delegate else {return}
+        present(delegate.showAlert(alertModel: alertModel), animated: true)
```

Comment:
так формально ок, хотя выглядит немного странно и не читаемо, предлагаю другой вариант
```suggestion
        guard let alert = delegate?.showAlert(alertModel: alertModel) else { return }
        present(alert, animated: true)
```

но судя по названию метода showAlert, тут логическая ошибка в декомпозии/ нейминге

контроллер держит презентер(delegate), и он умеет показывать алерт (showAlert), то есть предполагается, что презентер способен вызвать метод present. Делается либо через дополнительный параметр в функции, либо через явную ссылку в презентере на UIViewController